File: test/corpus/js/tricky/service.js
export class Service {
  constructor(repo) {
    this.repo = repo;
  }

  label(key) {
    const row = this.repo.find(key);
    if (row === null) return 'missing';
    return 'found:' + String(row.name).toUpperCase();
  }
}

export function collect(items, sink) {
  for (const item of items) {
    if (item > 0) sink(item);
  }
  return items.length;
}

File: test/corpus/python/tricky/service.py
class Repo:
    def find(self, key):
        raise NotImplementedError


class Service:
    def __init__(self, repo):
        self.repo = repo

    def label(self, key):
        row = self.repo.find(key)
        if row is None:
            return "missing"
        return "found:" + str(row["name"]).upper()


def collect(items, sink):
    for item in items:
        if item > 0:
            sink(item)
    return len(items)
